app/models/course_lesson.rb

```diff
@@ -24,16 +24,12 @@ def course_lesson_parts_in_order
   end
 
   def duration_in_minutes_in_words
-    number_of_hours = completion_time_in_minutes / 60
-    number_of_minutes = completion_time_in_minutes % 60
+    hours = completion_time_in_minutes / 60
+    minutes = completion_time_in_minutes % 60
 
-    hour_string = "hour".pluralize(number_of_hours)
-    minute_string = "minute".pluralize(number_of_minutes)
-
-    if number_of_hours.positive?
-      "#{number_of_hours} #{hour_string} #{number_of_minutes} #{minute_string}"
-    else
-      "#{number_of_minutes} #{minute_string}"
-    end
+    [].tap { |lesson_duration_string|
+      lesson_duration_string << "#{hours} hour".pluralize(hours)       unless hours.zero?
+      lesson_duration_string << "#{minutes} minute".pluralize(minutes) unless minutes.zero?
+    }.join(" ")
```

Comment:
`lesson_duration_string` is a lying variable name - really, it's `lesson_duration_parts`, it's an array, and you add one or two elements to it. Then you join them with " " as the join element.

I would rename it, and maybe rewriting it so it doesn't use the `tap` - I am not convinced it makes things any more readable.